be/tool/beck.c

```diff
@@ -1556,6 +1632,167 @@ static void genadd(uint64_t gen)
 	}
 }
 
+static int nv_scan_offset_init(uint64_t workers_nr,
+			       uint64_t partitions_nr)
+{
+	uint64_t  p;
+
+	M0_PRE(partitions_nr <= AO_NR);
+
+	m0_mutex_init(&off_info.oi_lock);
+	m0_mutex_lock(&off_info.oi_lock);
```

Comment:
Why we need lock here?

Reply:
There could be multiple workers trying to save offsets using 'nv_scan_offset_update()' at same time, this is protections against that.

Reply:
removed in commit #d09e8e3